src/pages/generalSettings/IntelligentAgentSettings.jsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+
+import { FormattedMessage } from 'react-intl';
+import { get, reduce, filter } from 'lodash-es';
+
+import { intelligentAgentSchema } from '../../constants/intelligentAgentSchema';
+import SettingsTextInput from '../../components/settings/SettingsTextInput';
+import Text from '../../components/Text';
+import DividerTitle from '../../components/DividerTitle';
+import Link from '../../components/Link';
+
+export default function IntelligentAgentSettings({
+  intelligentAgentSettingsFields,
+  currentValues,
+  setCurrentValues,
+  siteSettings,
+  setAllValid,
+}) {
+  return intelligentAgentSchema.map(intelligentAgent => {
+    const currentPlatformFields = get(
+      intelligentAgent,
+      ['data', 'fields'],
+      [],
+    );
+    const currentPlatformEnablingField = get(intelligentAgent, [
+      'data',
+      'enablingField',
+    ]);
+
+    const isCurrentPlatformEnabled = get(
+      currentValues,
+      currentPlatformEnablingField,
+      false,
+    );
+
+    const noCredsMissing =
+      intelligentAgentSettingsFields.filter(
+        currentField => get(currentValues, currentField) === '',
+      ).length === 0;
+
+    const allValid =
+      !isCurrentPlatformEnabled ||
+      (isCurrentPlatformEnabled && noCredsMissing);
+    setAllValid(allValid);
```

Comment:
if i'm reading correctly, this will cause setState to get called every time `IntelligentAgentSettings` is rendered. surprised there is no infinite loop here but either way this needs to be fixed